nav2_msgs/msg/PosesStamped.msg

```diff
@@ -0,0 +1 @@
+geometry_msgs/PoseStamped[] poses
```

Comment:
Is this really the most ergonomic way to handle this?

Reply:
Probably not, that's one main point I wanted to have discussed. The option I see are:

1- Leave PosesStamped.msg (or maybe the better-named PoseStampedArray.msg) and adapt where necessary (e.g. NavigateThroughPoses.action)
2- Create PoseStampedArray.msg in geometry_msgs
3- (ab)use nav_msgs/Path.msg and not create any new message type

From Nav2's perspective I guess #2 would be convenient but I don't know if @tfoote wants to open the door to creating Arrays from primitive data types (although there is already [one](https://github.com/ros2/common_interfaces/blob/rolling/geometry_msgs/msg/PoseArray.msg))

@SteveMacenski we probably need an intermediary solution to not block this PR for too long. Would you rather #1 (without any committal adaptation to  NavigateThroughPoses.action) or #3? Or do you see another option?

Reply:
I think there shouldn't be much if any objections to Option 2 -- can you file a ticket in `geometry_msgs` with a PR that implements and tag me?

Reply:
Done https://github.com/ros2/common_interfaces/pull/262/files